code/game/objects/items/weapons/chemical_flamethrower/chemical_flamethrower.dm

```diff
@@ -0,0 +1,262 @@
+		total_ammo += canister.ammo
+	if(total_ammo - amount < 0)
+		return FALSE
+
+	var/length = length(canisters)
+	var/difference = amount
+	for(var/i in 0 to length)
+		var/obj/item/chemical_canister/canister = canisters[length - i]
+		if(canister.ammo - difference <= 0)
+			difference -= canister.ammo
+			canister.ammo = 0
+		else
+			canister.ammo -= difference
+			difference = 0
+
+		if(!difference)
+			break
+
+	if(difference)
+		return FALSE
+	return TRUE
+
+/obj/item/chemical_flamethrower/extended
+	name = "extended capacity chemical flamethrower"
+	desc = "A flamethrower that accepts two chemical cartridges to create lasting fires."
+	max_canisters = 2
+
+/obj/item/chemical_flamethrower/extended/nuclear
+	name = "Syndicate extended capacity chemical flamethrower"
+	desc = "A flamethrower that accepts two chemical cartridges to create lasting fires. As black as the ash of your enemies."
+
+/obj/item/chemical_flamethrower/extended/nuclear/Initialize(mapload)
+	. = ..()
+	for(var/i in 1 to max_canisters)
+		canisters += new /obj/item/chemical_canister/extended/nuclear
+	update_canister_stats()
+
+/obj/item/chemical_canister
+	name = "chemical canister"
+	desc = "A simple canister of fuel. Does not accept any pyrotechnics except for welding fuel."
+	icon = 'icons/obj/tank.dmi'
+	icon_state = "oxygen"
+	container_type = OPENCONTAINER
+	/// How much ammo do we have? Empty at 0.
+	var/ammo = 100
+	/// Which reagent IDs do we accept
+	var/list/accepted_chemicals = list("fuel")
+	/// The burn temperature of our currently stored chemical
+	var/chem_burn_temp = T0C + 300
+	/// The burn duration of our currently stored chemical
+	var/chem_burn_duration = 10 SECONDS
+	/// How many firestacks will our reagent apply
+	var/fire_applications = 1
+	/// The currently stored reagent ID
+	var/current_reagent_id
+	/// How many units of the reagent do we need to have it's effects kick in?
+	var/required_volume = 10
+	/// Do we have a locked in reagent type?
+	var/has_filled_reagent = FALSE
+
+/obj/item/chemical_canister/Initialize(mapload)
+	. = ..()
+	create_reagents(50)
+
+/obj/item/chemical_canister/on_reagent_change()
+	if(has_filled_reagent)
+		visible_message("<span class='notice'>[src] doesn't accept any new chemicals!</span>")
+		return
+
+	if(!reagents.get_master_reagent_id() || !(reagents.get_master_reagent_id() in accepted_chemicals))
+		reagents.clear_reagents()
+		visible_message("<span class='notice'>[src] doesn't accept the most present chemical!</span>")
+		return
+
+	current_reagent_id = reagents.get_master_reagent_id()
+	reagents.isolate_reagent(current_reagent_id)
+	var/has_enough_reagents = reagents.total_volume >= required_volume
+	visible_message("<span class='notice'>[src] removes all reagents except for [current_reagent_id]. \
+					The reservoir has [reagents.total_volume] out of [required_volume] units. \
+					Reagent effects are [has_enough_reagents ? "in effect" : "not active"].</span>")
```

Comment:
is this visible to everyone? does everyone need to see what you're doing with it text wise?

Reply:
I saw it as something like a scanner blurting out some voicelines, but a `to_chat` works better prob